server/web/webx/type.go

```diff
@@ -0,0 +1,84 @@
+package webx
+
+import (
+	"github.com/beego/beego/v2/core/logs"
+	"github.com/beego/beego/v2/server/web/context"
+)
+
+// bizFunc is a function that wraps a business logic function with a context and parameter extraction function.
+type bizFunc[T any] func(ctx *context.Context, param T) (any, error)
+
+// extractFunc is a function that extracts parameters from the context.
+type extractFunc[T any] func(ctx *context.Context) (params T, err error)
+
+// Option options to T
+type Option[T any] func(ctx *context.Context, t *T) error
+
+// WrapperFromJson is a internalWrapper function for handling JSON in request's body.
+// It binds the JSON request body to the specified type T
+// See test cases for details
+func WrapperFromJson[T any](
+	biz bizFunc[T]) func(ctx *context.Context) {
+	return internalWrapper(biz, func(ctx *context.Context) (params T, err error) {
+		err = ctx.BindJSON(&params)
+		return
+	})
+}
+
+// WrapperFromForm is a internalWrapper function for handling form data in request.
+// It binds the form data to the specified type T
+// See test cases for details
+func WrapperFromForm[T any](
+	biz bizFunc[T]) func(ctx *context.Context) {
+	return internalWrapper(biz, func(ctx *context.Context) (params T, err error) {
+		err = ctx.BindForm(&params)
+		return
+	})
+}
+
+// Wrapper is use by beego ctx.Bind(any) api
+// It binds the data to the specified type T
+// See test cases for details
+func Wrapper[T any](
+	biz bizFunc[T],
+	opts ...Option[T]) func(ctx *context.Context) {
```

Comment:
remove `opts` from all functions. we don't need it. and then move the files to `server/web`, rename to `generic_wrapper`